src/test/java/com/marizueva/laboratory/hw2/ex2/LoginTestWithSoftAssertions.java

```diff
@@ -0,0 +1,243 @@
+                .id("user-name"));
+
+        softAssert.assertEquals(loggedUserName.getText(), "ROMAN IOVLEV");
+
+        // Click on "Service" subcategory in the header and check that drop down contains
+        // options. "Support, Dates, Complex Table, Simple Table, Tables With Pages,
+        // Different Elements"
+        WebElement headerItemService = driver.findElement(By
+                .cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8 a.dropdown-toggle"));
+        headerItemService.click();
+
+        WebElement subMenuFromHeaderSupportItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li/"
+                        + "a[text()='Support']"));
+        softAssert.assertTrue(subMenuFromHeaderSupportItem.isDisplayed());
+        WebElement subMenuFromHeaderDatesItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li/a[text()='Dates']"));
+        softAssert.assertTrue(subMenuFromHeaderDatesItem.isDisplayed());
+        WebElement subMenuFromHeaderSearchItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li/a[text()='Search']"));
+        softAssert.assertTrue(subMenuFromHeaderSearchItem.isDisplayed());
+
+        WebElement subMenuFromHeaderComplexTableItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li"
+                        + "/a[@href='complex-table.html']"));
+        softAssert.assertTrue(subMenuFromHeaderComplexTableItem.isDisplayed());
+        WebElement subMenuFromHeaderUserTableItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li"
+                        + "/a[@href='user-table.html']"));
+        softAssert.assertTrue(subMenuFromHeaderUserTableItem.isDisplayed());
+        WebElement subMenuFromHeaderTableWithPagesItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li"
+                        + "/a[@href='table-pages.html']"));
+        softAssert.assertTrue(subMenuFromHeaderTableWithPagesItem.isDisplayed());
+        WebElement subMenuFromHeaderDifferentElementsItem = driver.findElement(
+                By.xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']"
+                        + "/li/a[@href='different-elements.html']"));
+        softAssert.assertTrue(subMenuFromHeaderDifferentElementsItem.isDisplayed());
```

Comment:
Looks like, that here you could get List<WebElement> for the menu items and thought the loop check that all they displayed instead of getting each element from the page.
It increases speed of your test

Reply:
fixed

Reply:
some excess assertions were removed